### DISCUSSION TITLE
Feature Request：about keyword var

### DISCUSSION THREAD

**Discussion by rainryu** [dotnet/csharplang#3569] [Language Ideas]
Version Used:
c#8.0

Steps to Reproduce:
if i want write like this

**var configurationRoot = configurationBuilder.Build();**

sometimes im not sure return type ;

Expected Behavior:
i hope i can write like this
var configurationBuilder.Build() = configurationRoot ；

Actual Behavior:
i write a balbalbl name like xxx first

**var _xxx_= configurationBuilder.Build();**

when is write .build, oh i get it and return to xxx,change it to right name like configurationRoot .

**Comment by HaloFour** [comment]
Could always write the expression and then use the [Introduce Local](https://docs.microsoft.com/en-us/visualstudio/ide/reference/introduce-local-variable?view=vs-2019) quick action to add a local.  Or I'd suggest naming the variable after what it does or what it means to the program rather than based on it's type.  IMO `configuration` would be a better name than `configurationRoot` anyway.

**Comment by svick** [comment]
Previous discussion: https://github.com/dotnet/roslyn/issues/45169.

**Comment by bbarry** [comment]
This isn't really a language issue but a tooling one.

With Resharper I would type:

    cB.B.v<tab><tab>configurationRoot

I'm so used to R# I don't entirely know what basic VS stuff could do in a similar way. Probably something like:

    configuration<tab>.<down><tab><right><right><shift+home><ctrl+.><select introduce local>configurationRoot

**Comment by Pzixel** [comment]
use

```cs
if (configurationBuilder.Build() is var configurationRoot) 
{
}
```

or write an extension

```cs
public static class ObjectExtensions
{
    public static void Then<T>(this T obj, Action<T> func)
    {
        if (obj == null)
        {
            throw new ArgumentNullException(nameof(obj));
        }

        if (func == null)
        {
            throw new ArgumentNullException(nameof(func));
        }
        func(obj);
    }
    
    public static TResult Then<T, TResult>(this T obj, Func<T, TResult> func) 
    {
        if (obj == null)
        {
            throw new ArgumentNullException(nameof(obj));
        }

        if (func == null)
        {
            throw new ArgumentNullException(nameof(func));
        }
        return func(obj);
    }        
}
```

Then use:

```cs
configurationBuilder.Build().Then(configurationRoot => ...)
```
